addons/web/static/src/core/datetime/datetime_picker.xml

```diff
@@ -123,24 +122,24 @@
 
     <t t-name="web.DateTimePicker" owl="1">
         <div
-            class="o_datetime_picker bg-view d-flex flex-column p-3 gap-2 user-select-none"
+            class="o_datetime_picker rounded bg-view d-flex flex-column p-3 gap-2 user-select-none"
             t-attf-style="--DateTimePicker__Day-template-columns: {{ isRange ? 7 : 8 }}"
         >
             <nav class="o_datetime_picker_header btn-group">
-                <button class="o_previous btn btn-secondary flex-grow-0" t-on-click="previous">
+                <button class="o_previous btn btn-light flex-grow-0" t-on-click="previous">
                     <span class="oi oi-chevron-left" t-att-title="activePrecisionLevel.prevTitle" />
                 </button>
                 <button
-                    class="o_zoom_out o_datetime_button btn d-flex align-items-center justify-content-around text-truncate opacity-100"
+                    class="o_zoom_out o_datetime_button btn btn-light d-flex align-items-center justify-content-around px-0 text-truncate opacity-100"
                     t-att-disabled="isLastPrecisionLevel"
                     t-att-title="!isLastPrecisionLevel and activePrecisionLevel.mainTitle"
                     t-on-click="zoomOut"
                 >
                     <t t-foreach="titles" t-as="title" t-key="title">
-                        <h5 class="o_header_part fw-bold m-0" t-esc="title" />
+                        <h5 class="o_header_part fw-bold m-0 w-50" t-esc="title" />
```

Comment:
Since we want to customise the font-weight, I propose
```suggestion
                        <strong class="o_header_part w-50 fs-5" t-esc="title" />
```

Also, I would apply `w-50` only if:
- Screen => LG
- Both tables are rendered

...so we let the title expand when it's possible. Consider using `flex-basis-x`utility classes, to be tested  ;)